pkg/serviceprovider/gitlab/gitlab.go

```diff
@@ -64,6 +68,63 @@ type Gitlab struct {
 	downloadFileCapability downloadFileCapability
 }
 
+func (g Gitlab) RefreshToken(ctx context.Context, token *api.Token, clientId string, clientSecret string) (*api.Token, error) {
+	lg := log.FromContext(ctx)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost,
+		g.baseUrl+"/oauth/token?client_id="+clientId+
```

Comment:
I think I would prefer here `fmt.Sprintf` or even construct it with `url.URL`=>`url.String()` over concatenation. But personal preference.

Reply:
Done.